src/com/amazon/ionschema/Import.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema
+
+/**
+ * An Import represents all the types imported by a Schema
+ * from one schema id.
+ *
+ * Note that multiple ISL imports referencing the same schema id
+ * (each importing/aliasing an individual type) are represented by a
+ * single Import instance.
+ */
+interface Import {
+    /**
+     * The id of the referenced schema.
+     */
+    val id: String
+
+    /**
+     * Returns the schema referenced by the import.
+     */
+    fun getSchema(): Schema
+
+    /**
+     * Returns the requested type, if present in this import;
+     * otherwise returns null.  If a type is aliased (via "as")
+     * by the import, the type will only be returned from this
+     * method by its alias (not the imported type's original name).
+     */
+    fun getType(name: String): Type?
```

Comment:
Given that we have `getType(name:String)` it will be straight forward to implement a `containsType(name:String): bool` predicate. Does it make it easier to provide that method as well on this interface?

Reply:
My sense is that `containsType(key)` is not a great pattern.  I suspect typical usage of that method would be followed by `getType(key)` -- at which point, you've performed two look-ups instead of just one (plus a null check).

Reply:
This is not a blocker and I believe touches on personal preference on APIs. 

Well, if there is a  `containsType` method then the next step is to make the return type of `getType` be `Type` rather than `Type?`. If people call `getType` without first using `containsType` then you can throw.  We still have 2 lookups which may be a cost, but, we no longer have `null` as a possible return value and a check for `null` which may or may not happen on the client side probably more than once (if I chain calls or pass the return value around). 

I may be a little too eager in removing `null` from the API. As I said, not blocking, mostly exploring the API design space.

Reply:
Yeah, feels too eager to me.

The nice thing is that checking for null after calling `getType(name): Type?` is optional.  If you don't check in Java, you may wind up with an NPE.  In Kotlin, simply append a not-null assertion (`!!`).

Leaving as-is.  We can narrow the return type (to exclude `null`) later if desired, but there's something to be said for following the pattern of platform APIs like `Map.get(key: K): V?`.